internal/kafka/internal/handlers/validation.go

```diff
@@ -145,18 +208,19 @@ func ValidateKafkaUserFacingUpdateFields(ctx context.Context, authService author
 	}
 }
 
-func ValidateKafkaClaims(ctx context.Context, kafkaRequestPayload *public.KafkaRequestPayload, kafkaRequest *dbapi.KafkaRequest) handlers.Validate {
-	return func() *errors.ServiceError {
-		kafkaRequest = presenters.ConvertKafkaRequest(*kafkaRequestPayload, kafkaRequest)
-		claims, err := auth.GetClaimsFromContext(ctx)
-		if err != nil {
-			return errors.Unauthenticated("user not authenticated")
-		}
-		(*kafkaRequest).Owner = auth.GetUsernameFromClaims(claims)
-		(*kafkaRequest).OrganisationId = auth.GetOrgIdFromClaims(claims)
-		(*kafkaRequest).OwnerAccountId = auth.GetAccountIdFromClaims(claims)
+func getClaims(ctx context.Context) (jwt.MapClaims, *errors.ServiceError) {
+	claims, err := auth.GetClaimsFromContext(ctx)
+	if err != nil {
+		return nil, errors.Unauthenticated("user not authenticated")
+	}
+	return claims, nil
+}
 
-		return nil
+// ValidateKafkaClaims - Verifies that the context contains the required claims
+func ValidateKafkaClaims(ctx context.Context) handlers.Validate {
+	return func() *errors.ServiceError {
+		_, err := getClaims(ctx)
+		return err
 	}
```

Comment:
What claims are being validated? I think this can be removed all together if we do not validate specific claims

Reply:
It ensure that the request contains the claims.

Reply:
> It ensure that the request contains the claims.

...but this is done in other validators already that rely on claims.
 
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f26d8f9fce7075e3b309928fdfe8bf24eca8bdc1/internal/kafka/internal/handlers/validation.go#L79-L82
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/f26d8f9fce7075e3b309928fdfe8bf24eca8bdc1/internal/kafka/internal/handlers/validation.go#L116-L119

Reply:
true, but that is only by incident, because those validators needs authentication.
If we want to be purist in the rule design pattern, then each rule should make one check. That means that one check will be used to check for claims to be present, while the other checks will be relied upon only for their respective responsibilities.

The other two checks you linked are responsible of checking the provider and region values and the instance type and size values. Relying on those for checking the claims presence would break the rule design pattern contract.

Reply:
> true, but that is only by incident, because those validators needs authentication.

You nailed it. This is where I wanted to arrive.. to me the validation as it is at the moment, seems to be performing authentication related validation and I think we have a middleware for that. Is this correct?

Reply:
yup. And I don't think the validators should rely upon the presence or not of such middleware.

Reply:
> I think the validators should check that everything that is needed by the handler is present and is valid.

+1

> So, if the handler needs the claims to be present, then the validators should check for the claims to be present

+1 on this if we performed specific validation on specific claims, which is not the case for the moment. And the validator is reduced to performing a role which is already performed at a layer above (the middleware).

Reply:
Added options to the validator so that the claims validator can be configured enabling or disabling validations.
Mind giving another look?

Reply:
Thanks. Now the validation makes sense.